common/src/main/java/org/opensearch/ml/common/memorycontainer/MemoryStrategy.java

```diff
@@ -130,13 +131,44 @@ public static MemoryStrategy parse(XContentParser parser) throws IOException {
             }
         }
 
-        // Generate ID with type prefix if not provided
-        if (id == null) {
-            id = generateStrategyId(type);
-        }
+        // Do not auto-generate ID during parsing - let StrategyMergeHelper control ID generation
+        // When id is null/empty, StrategyMergeHelper will detect it as a new strategy and generate ID
         return MemoryStrategy.builder().id(id).enabled(enabled).type(type).namespace(namespace).strategyConfig(strategyConfig).build();
     }
 
+    /**
+     * Returns whether this strategy is enabled. If enabled is null, defaults to true.
+     *
+     * @return true if enabled or null, false otherwise
+     */
+    public boolean isEnabled() {
+        return enabled == null || enabled;
+    }
+
+    /**
+     * Validates a memory strategy for required fields and valid values.
+     *
+     * @param strategy The memory strategy to validate
+     * @throws IllegalArgumentException if validation fails with specific error message
+     */
+    public static void validate(MemoryStrategy strategy) {
```

Comment:
This validation isn't automatically performed in constructor method, meaning invalid objects could be created if the calling code doesn't explicitly call the validate() method.